### PR TITLE
Attempt at porting StrictDeepEqual Issue#662

### PR DESCRIPTION
**Pull request by standvpmnt**
#662 Draft PR to get help and ideas from maintainers about work here. [Issue#662](https://github.com/denoland/deno_std/issues/662)

### CODE REVIEW

**Comment by standvpmnt** [comment]
Is there a way for me to block the CI tests from running as I already know that they are failing?

**Comment by bartlomieju** [comment]
> Is there a way for me to block the CI tests from running as I already know that they are failing?

No, but it's not a big deal

File: node/comparisons_test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import { assertEquals } from "../testing/asserts.ts";
+import { isDeepStrictEqual } from "./comparisons.ts";
+
+Deno.test("for basic Objects", async (t) => {
```
Comment (kt3k) [inline node/comparisons_test.ts:4]: These test names share the same namespace with other test cases in std library. So can you prefix the case name like `[node/comparison] for basic Objects`? (I don't think we need such prefixes for sub test steps as they are already grouped)

Comment (standvpmnt) [reply node/comparisons_test.ts]: Sure, I have updated these with the namespace

File: node/comparisons.ts [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+          return false;
+        }
+      }
+      // else if (
+      //   !strict &&
+      //   // (!map1.has(key) || !isDeepEqual(map1.get(key), item, false, memos)) &&
+      //   // !mapHasEqualEntry(set, map1, key, item, false, memos)
+      // ) {
+      //   return false;
+      // }
```
Comment (kt3k) [inline node/comparisons.ts:565]: ditto. please leave some TODO comment about this commented-out block, or remove them

File: node/comparisons.ts [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+
+  return true;
+}
+
+// function setMightHaveLoosePrim(set1, set2, primitive) {
+//   const altValue = findLooseMatchingPrimitives(primitive);
+//   if (altValue != null) return altValue;
+
+//   return set2.has(altValue) && !set1.has(altValue);
+// }
```
Comment (kt3k) [inline node/comparisons.ts:578]: ditto. please leave some TODO comment about this commented-out block, or remove them

File: node/comparisons.ts [hunk 4]
```diff
@@ -0,0 +1,619 @@
...
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// function compare(a: Uint8Array, b: Uint8Array): number {
+//   return 0;
+// }
```
Comment (kt3k) [inline node/comparisons.ts:619]: ditto. please leave some TODO comment about this commented-out block, or remove them

**Review by kt3k — COMMENTED** [review]
Looks mostly good to me! Left some comments about minor code organization points.

Thank you for working on this!

**Comment by kt3k** [comment]
BTW It'd be helpful if you would move `node/comparisons.ts` to `node/internal/util/comparisons.ts`. We'd like to keep the dir structure of `node/` as similar as possible to the dir structure of `lib/` in node.js repository.

**Review by kt3k — APPROVED** [review]
@standvpmnt LGTM. Thank you for working on this for a long time!

**Comment by standvpmnt** [comment]
Thanks a lot for being so patient and all the inputs with this PR @kt3k I know wasn't able to work dedicatedly like I had anticipated when I took on this PR, none the less I am glad I could contribute a little to deno. And thanks @bartlomieju for putting the labels on the issues that really motivated me to get started with this and I definitely learned a fair bit about JS internals by working on this.

**Pull request merged**